### PR TITLE
Ability to add products to unconfirmed orders

### LINKED ISSUE #4099: [Feature]: Ability to add products to unconfirmed orders
### Description of the feature

User should be able to change product in an unconfirmed orders, the same way that they can on a draft orders, since unconfirmed orders aren't fixed by definition. The customer can call and ask for changes to the line item list.

### Acceptable criteria

- [ ] As a user, when edit an unconfirmed order, I should be able to add new products to it.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by lephuongbg**
closes #4099

Basically I copy pasted code from draft order implementation over to unconfirmed order page.

### Screenshots

Before

image

After

image

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `data-test-id` are added for new elements
4. [x] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [x] Your code works with the latest stable version of the core
6. [x] I added changesets and [read good practices](../.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

File: .changeset/kind-apes-look.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"saleor-dashboard": patch
```
Comment (andrzejewsky) [inline .changeset/kind-apes-look.md:2]: since it's a new feature, it should be "minor"

**Review by Droniu — DISMISSED** [review]

**Review by andrzejewsky — CHANGES_REQUESTED** [review]
Thanks for PR!, I left one comment.

**Comment by andrzejewsky** [comment]
@lephuongbg Hey any update on it? Could you apply mentioned comment?

**Comment by lephuongbg** [comment]
@andrzejewsky Thanks for the nudge. Updated the PR.

**Comment by andrzejewsky** [comment]
@lephuongbg Hey I was reviewing your PR including some UI checks, seems like it causes an error when I try to add the product:


Screenshot 2023-09-27 at 11 17 07

**Comment by lephuongbg** [comment]
@andrzejewsky Can I have the detailed graphql error response that you met?

**Comment by wojteknowacki** [comment]
Hello @lephuongbg . Error message mentioned by @andrzejewsky :
`{
  "errors": [
    {
      "message": "Variable \"$isStaffUser\" of required type \"Boolean!\" was not provided.",
      "locations": [
        {
          "line": 1,
          "column": 34
        }
      ],
      "extensions": {
        "exception": {
          "code": "GraphQLError"
        }
      }
    }
  ]
}`

**Pull request closed without merging**